lnwallet/wallet.go

```diff
@@ -821,42 +821,14 @@ func (l *LightningWallet) initOurContribution(reservation *ChannelReservation,
 		return err
 	}
 
-	// With the above keys created, we'll also need to initialization our
-	// initial revocation tree state.
-	nextRevocationKeyDesc, err := keyRing.DeriveNextKey(
-		keychain.KeyFamilyRevocationRoot,
-	)
-	if err != nil {
-		return err
-	}
-
-	// If the DeriveNextKey call returns the first key with Index 0, we need
-	// to re-derive the key as the keychain/btcwallet.go DerivePrivKey call
-	// special-cases Index 0.
-	if nextRevocationKeyDesc.Index == 0 {
-		nextRevocationKeyDesc, err = keyRing.DeriveNextKey(
-			keychain.KeyFamilyRevocationRoot,
-		)
-		if err != nil {
-			return err
-		}
-	}
-
-	reservation.nextRevocationKeyLoc = nextRevocationKeyDesc.KeyLocator
-
-	// Perform an ECDH operation between the private key described in
-	// nextRevocationKeyDesc and our public multisig key. The result will be
-	// used to seed the revocation producer.
-	revRoot, err := l.ECDH(
-		nextRevocationKeyDesc, reservation.ourContribution.MultiSigKey.PubKey,
-	)
+	// With the above keys created, we'll also need to initialize our
+	// revocation tree state, and from that generate the
+	// per-commitment point.
+	producer, err := l.nextRevocationProducer(reservation, keyRing)
```

Comment:
nit: extract this in the commit where we first change the code in this file? Or even in a separate commit before that?